extensions/rules_haskell_dependencies.bzl

```diff
@@ -23,9 +24,27 @@ def repositories(*, bzlmod):
 
     # TODO: Remove when tests are run with a ghc version containing Cabal >= 3.10
     # See https://github.com/tweag/rules_haskell/issues/1871
-    http_archive(
-        name = "Cabal",
-        build_file_content = """
+    if GHC_VERSION and GHC_VERSION.startswith("9.4."):
```

Comment:
So long as it's only for our CI, I'm fine with this approach. Generally speaking, a better route would be if we could figure out how to `select` based on resolved GHC version and then use select to switch to the correct versions. But, that seems quite difficult in this case.